### ISSUE TITLE
Threads with names of "MessageExchangeStore1" and "ObservationStore1" are left behind after LeshanClient.destroy is called

### ISSUE THREAD

**Issue by lhotari** [eclipse/leshan#502]
Version: Leshan 1.0.0-M6

There is a problem that a Leshan client doesn't shutdown cleanly. It leaves threads with names "MessageExchangeStore1" and "ObservationStore1" behind. 
These threads are created in Californium Core, in [InMemoryMessageExchangeStore](https://github.com/eclipse/californium/blob/d3ab69254d4948a1a2a9a798197cb34a35810682/californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java#L124) and [InMemoryObservationStore](https://github.com/eclipse/californium/blob/d3ab69254d4948a1a2a9a798197cb34a35810682/californium-core/src/main/java/org/eclipse/californium/core/observe/InMemoryObservationStore.java#L55).

These threads don't get created when the HEALTH_STATUS_INTERVAL setting is 0. This can be used as a workaround:
```
networkConfig.setInt(NetworkConfig.Keys.HEALTH_STATUS_INTERVAL, 0);
```

The expected behaviour is that all resources get closed with LeshanClient.destroy is called.

**Comment by sbernard31** [comment]
Those 2 threads help to monitor `InMemoryMessageExchangeStore` and `InMemoryObservationStore`, this is for debug only.

This should not be activated by default. This is the case in Californium 2.0.0-M9 which is a "bug" and this should be fixed in the next milestones.

I will force it as deactivated by default in Leshan `master` waiting we integrate the next californium release.

Ideally, even if this is a debug feature only the thread should be destroyed on LeshanClient.destroy.
This would be easy to do that for `InMemoryMessageExchangeStore` but not so easy for `InMemoryObservationStore`, so not sure this will be done.

**Comment by sbernard31** [comment]
Another workaround is to set logger below to WARN, ERROR or OFF : 
`org.eclipse.californium.core.network.InMemoryMessageExchangeStore.health`
`org.eclipse.californium.core.observe.InMemoryObservationStore.health`

**Comment by sbernard31** [comment]
This should be workarounded in master. (commit e438d74)

**Comment by sbernard31** [comment]
A fix was proposed in californium : https://github.com/eclipse/californium/pull/657

**Comment by sbernard31** [comment]
The fix is integrated in californium 2.0.x and should be available in the next release.
We have a workaround in Leshan too, we can close this issue.
